src/object.c

```diff
@@ -380,27 +382,240 @@ int object_calculate_checksum(const rct_object_entry *entry, const char *data, i
+		RCT2_GLOBAL(RCT2_ADDRESS_G1_ELEMENTS + esi + 4, uint32_t) = ((uint32_t*)(edx + (*ebp)))[1];
+		RCT2_GLOBAL(RCT2_ADDRESS_G1_ELEMENTS + esi + 8, uint32_t) = ((uint32_t*)(edx + (*ebp)))[2];
+		RCT2_GLOBAL(RCT2_ADDRESS_G1_ELEMENTS + esi + 0xC, uint32_t) = ((uint32_t*)(edx + (*ebp)))[3];
+		RCT2_GLOBAL(RCT2_ADDRESS_G1_ELEMENTS + esi, uint32_t) += ebx;
+		esi += 0x10;
+		edx += 0x10;
+		ecx--;
+		if (ecx == 0) break;
+	}
+	ebx = ((uint32_t*)((*ebp) - 8))[0];
+	ebx *= 0x10;
+	(*ebp) += ((uint32_t*)((*ebp) - 4))[0];
+	(*ebp) += ebx;
+	return result;
+}
+
+int paint_path_entry(int flags, int ebx, int ecx, int edx, rct_drawpixelinfo* dpi, int esi, int ebp)
+{
+	if ((flags & 0xFF) != 3)
+	{
+		if ((flags & 0xFF) != 1)
+		{
+			if ((flags & 0xFF) <= 1)//0
+			{
+				uint8_t* ebp = esi + 0xE;
+				((rct_string_id*)esi)[0] = object_get_localised_text(&ebp, ecx, ebx, 0);
+				int a = sub_6A9ED1(&ebp);
+				((uint32_t*)(esi + 2))[0] = a;
+				a += 0x6D;
+				((uint32_t*)(esi + 6))[0] = a;
+				if (RCT2_GLOBAL(0x9ADAF4, uint32_t) != 0xFFFFFFFF) RCT2_GLOBAL(0x9ADAF4, uint16_t*)[0] = 0;
+				RCT2_GLOBAL(RCT2_ADDRESS_SELECTED_PATH_ID, sint16) = 0;
+				int b = -1;
+				while (true)
+				{
+					b++;
+					if (b >= 0x10) break;
+					uint8_t* edi = object_entry_groups[5].chunks[ebx];
+					if (edi == 0xFFFFFFFF) continue;
+					if (!(edi[0xB] & 4))
+					{
+						RCT2_GLOBAL(RCT2_ADDRESS_SELECTED_PATH_ID, sint16) = ebx;
+						break;
+					}
+					RCT2_GLOBAL(RCT2_ADDRESS_SELECTED_PATH_ID, sint16) = ebx;
+				}
+				return flags;
+			}
+			else
+			{
+				if (((uint8_t*)(esi + 0xA))[0] >= 2) return 1;//actually the carry bit should be set (stc)
+				else return 0;
+			}
+		}
+		else
+		{
+			((rct_string_id*)esi)[0] = 0;
+			((uint32_t*)(esi + 2))[0] = 0;
+			((uint32_t*)(esi + 6))[0] = 0;
+			return flags;
+		}
+	}
+	else
+	{
+		if (!((flags >> 8) & 0xFF))
+		{
+			//Draws preview for scenario editor!
+			int b = ((uint32_t*)(ebp + 2))[0];
+			b += 0x47;
+			ecx -= 0x31;
+			edx -= 0x11;
+			gfx_draw_sprite(dpi, b, ecx, edx, ebp);
+			b++;
+			ecx += 0x35;
+			gfx_draw_sprite(dpi, b, ecx, edx, ebp);
+		}
+		return flags;
+	}
+}
+
+int paint_park_entrance_entry(int flags, int ebx, int ecx, int edx, rct_drawpixelinfo* dpi, int esi, int ebp)
+{
+	if ((flags & 0xFF) != 3)
+	{
+		if ((flags & 0xFF) != 1)
+		{
+			if ((flags & 0xFF) <= 1)//0
+			{
+				uint8_t* ebp = esi + 8;
+				((rct_string_id*)esi)[0] = object_get_localised_text(&ebp, ecx, ebx, 0);
+				int a = sub_6A9ED1(&ebp);
+				((uint32_t*)(esi + 2))[0] = a;
+				if (RCT2_GLOBAL(0x9ADAF4, uint32_t) != 0xFFFFFFFF) RCT2_GLOBAL(0x9ADAF4, uint16_t*)[0] = 0;
+				return flags;
+			}
+			else
+			{
+				return 0;
+			}
+		}
+		else
+		{
+			((rct_string_id*)esi)[0] = 0;
+			((uint32_t*)(esi + 2))[0] = 0;
+			return flags;
+		}
+	}
+	else
+	{
+		if (!((flags >> 8) & 0xFF))
+		{
+			dpi = clip_drawpixelinfo(dpi, ecx - 0x38, 0x70, edx - 0x38, 0x70);
```

Comment:
Try to avoid using hexidecimal for these as its easier to read. I tend to use hexadecimal only if calling an unknown function or it involves flags.